vllm/v1/core/kv_cache_manager.py

```diff
@@ -200,6 +200,11 @@ def allocate_slots(
             raise ValueError(
                 f"num_tokens must be greater than 0, got {num_tokens}")
 
+        num_required_blocks = cdiv(num_tokens, self.block_size)
+        if (num_required_blocks > self.free_block_queue.num_free_blocks):
```

Comment:
If moving the checker to here, the free block number should minus the computed blocks (note that computed blocks may or may not in the free queue so we cannot directly minus len(computed_blcoks).